### DISCUSSION TITLE
fanboy's annoyance: general vs ubo flavor

### DISCUSSION THREAD

**Discussion by ghost** [uBlockOrigin/uBlock-issues#2612] [Q&A] [answered]
All settings being the same and filterlists updated just now the general fanboy's annoyance currently uses roughly 76k/76k for me but the ubo one uses 76k/91k, so there are 15k new filters which aren't used.
Why is that so, where does the overlap come from and why is this beneficial?

Related issue: #2610

**Comment by ghost** [comment, marked as answer]
![Screenshot_20230426_211511](https://user-images.githubusercontent.com/886325/234679634-08fc0170-ab64-4ea4-87c6-892a23c82deb.png)
![Screenshot_20230426_211518](https://user-images.githubusercontent.com/886325/234679637-9d8755bc-e762-438d-ba6c-7f580d8dbe57.png)

**Reply by gorhill** [reply]
So in the end there was no point to <https://github.com/uBlockOrigin/uBlock-issues/issues/2610>?

**Reply by gorhill** [reply]
Ok there was a point to #2610, the demonstration was that the following filter could not be found in `https://secure.fanboy.co.nz/fanboy-annoyance.txt`:

    similarweb.com##+js(rc, app-banner-parent, body, stay)

It can be found in `https://secure.fanboy.co.nz/fanboy-annoyance_ubo.txt`.

Also it appears `https://secure.fanboy.co.nz/fanboy-cookiemonster_ubo.txt` suffers the same duplicated blocks of filters issue.

**Comment by ghost** [comment]
Should this be reported to the filterlist maintainer? 15k of duplicates seems excessive to me.

**Reply by gorhill** [reply]
See <https://github.com/uBlockOrigin/uAssets/commit/29d8aa1c4abc2dfd3d38a9a5dada0360d53898c2>.

**Comment by peace2000** [comment]
As this topic was brought up, I have also noticed that there doesn't seem to be a way to subscribe to Fanboy's Newsletter or Fanboy's Notifications in a way that their uBO additions would be included as well.

In case I want to use those lists separately, without the Full Annoyance set, I'll have to add their uBO additions manually from Github this way:

![kuva](https://user-images.githubusercontent.com/17256841/234911932-9d5b3fea-e0f5-49d6-9be8-b99fc0dead13.png)

Could you @gorhill consider adding Fanboy's Newsletter and Notifications to uBO, in a way that their uBO additions would be included as well?

Not sure if you want to add more separate Annoyance based list entries to uBO, though the Social list & Cookie list are there already as well as a separate entries from the Full Annoyance list (I.E. why Social & Cookie list should be there as a separate entries but not Newsletter or Notifications?).

I use these urls as sources for those lists in the picture:

`https://easylist.to/easylist/fanboy-newsletter.txt`
`https://raw.githubusercontent.com/easylist/easylist/master/fanboy-addon/fanboy_newsletter_specific_uBO.txt`

`https://easylist-downloads.adblockplus.org/fanboy-notifications.txt`
`https://raw.githubusercontent.com/easylist/easylist/master/fanboy-addon/fanboy_notifications_specific_uBO.txt`

Personally, I have no issues using this "clumsy" way, but I'm afraid that some uBO users might not notice that uBO additions are missing.

**Reply by gorhill** [reply]
I am not convinced that _newsletters_ and _notifications_ should be spin-off  from _annoyances_ -- I think most people would put these categories as annoyances and unlikely to pick one and not the others, unlike _cookies_ or _social_.

**Reply by gorhill** [reply]
If we spin-off _newsletters_ and _notifications_, what's left in _annoyances_? How what is left there can be described?

**Reply by krystian3w** [reply]
And so you would probably have to teach people what is it a list hosted under uBo, because I see only 11 k filters - that is, people will have to activate two more lists and explain that they are no longer combined into one.

![image](https://user-images.githubusercontent.com/35370833/235075720-fb4e2d19-f056-4f4b-ab60-83a8f40403e1.png) |
--- |

**Reply by Yuki2718** [reply]
it's long been a problem and I brought up in Slack.
> I am not convinced

I was one who subscribed Notification list standalone. Just the same reason as AG splitted Annoyances, there is actual needs to use only sublist but not whole Annoyance list which tends to block too much. I'm no more interested in the Notification list as it started to block mobile app banners too which IMO should be handled by another sublist.

**Reply by Yuki2718** [reply]
> what's left in annoyances?

Not a user of FA but there are many, from right-click prevention to snowfall to all sort of popup not falling down to newsletter. There are also promotions and some videos.

**Reply by peace2000** [reply]
> If we spin-off _newsletters_ and _notifications_, what's left in _annoyances_? How what is left there can be described?

Well, the "Annoyances" section that is without clear categorization... It's not easy to describe. I would call it miscellaneous or other. This is the issue with Annoyance lists that they are somewhat vague always, unless there are clearly categorized sections.

One thing to consider could be removing separate entries for the Easylist Cookie and Social list in the Annoyance category and just put them under Fanboy's Annoyace as a separate, selectable entries. And the same for other categories that belong to Fanboy's Annoance.

https://user-images.githubusercontent.com/17256841/235269628-f0a2367b-e8bc-4f18-8ddf-e2178a5dbcaf.png

This would give users clear and easy choice what they want to block. Also many people are unaware that Easylist Cookie or Social list are included in Fanboy Annoyance full list. This step would also clear that up.

**Reply by gothic-bum** [reply]
Perhaps an "annoyances" list shouldn't be included at all, since it is entirely subjective what is annoying or not ... unless it can somehow be objectively defined (which AFAIK it can't).
  
This is in contrast to "newsletters", "notifications," "social", and "cookie",  which are all more specific and can be more clearly defined.
  
So "annoyances" could be considered a category rather than a list, which then contains individual lists for "newsletters", "notifications," "social", "cookie", etc.

**Reply by krystian3w** [reply]
Remember implement in same time modules for AdGuard #2217 if we implement tree/nesting in Annoyances.

> _**(Draft with MD check list)**_
> 
> - [ ] AG Annoyances
>    - [ ] Cookie Notices
>    - [ ] Popups
>    - [ ] Mobile app banners
>    - [ ] Widgets
>    - [ ] Others Annoyances (undefined in other categories)

**Reply by Yuki2718** [reply]
> Also many people are unaware that Easylist Cookie or Social list are included in Fanboy Annoyance full list.

Yes, we still see people subscribe EC or FS despite FA is already subscribed.

**Comment by ghost** [comment]
https://github.com/gorhill/uBlock/commit/e6e5b1eac43ba6f2408ad7957c7671b809808918 didn't change the old general flavor social list link. The ubo flavor one is `https://secure.fanboy.co.nz/fanboy-social_ubo.txt`.

**Reply by peace2000** [reply]
@c5aba924-b91a-4504-b8e7-9dd81a704703 gorhill haven't released a version of uBO yet where that new url is put into use. This Annoyance thing might still be under consideration, how this should be handled as a whole.

**Reply by gorhill** [reply]
> didn't change the old general flavor social list link

Give it time, changes to lists take days to propagate.

**Reply by ghost** [reply]
Sorry for my bad english.
I just tried to report an error in the above mentioned commit: This [link](https://github.com/gorhill/uBlock/commit/e6e5b1eac43ba6f2408ad7957c7671b809808918#diff-867c526785ad7b2296cb4d7ad5a069232add7ad772faf27beba1ce9570f288abL302) wasn't updated to `https://secure.fanboy.co.nz/fanboy-social_ubo.txt`, so currently one _randomly_ get the general or ubo flavor.

**Reply by gorhill** [reply]
~~I will update the link, but~~ those `contentURL` links are not picked randomly, they are picked by order of appearance and if one URL fails to fetch the list, the next one is used. The `cdnURLs` links are picked randomly.

---

Actually I don't need to update the link, there is an `!#include fanboy_social_specific_uBO.txt` in there.

**Reply by ghost** [reply]
> Actually I don't need to update the link, there is an `!#include fanboy_social_specific_uBO.txt` in there.

Now I'm confused.
There is also a `!#include easylist_cookie_specific_uBO.txt` in the general flavor `https://secure.fanboy.co.nz/fanboy-cookiemonster.txt`.

**Reply by gorhill** [reply]
You shouldn't worry about `https://secure.fanboy.co.nz/...` lists, they are meant to be used as fallback only, i.e. in extremely rare occasion, if ever.

**Reply by krystian3w** [reply]
:us: But the list of annoyances "from" fanboy.co.nz should get obsolete status, as you do not paste into the "EasyList Annoyances" cutting out social and cookies.

:uk: But the Annoyances list "from" fanboy.co.nz  should get obsolete status, as you don't paste into the 'EasyList Annoyances' cutting out social and cookies.

**Reply by gorhill** [reply]
Yes, of course I know this, I am just taking my time with all this and pondering whether I go ahead with splitting into subgroup, not trivial work.

**Comment by peace2000** [comment]
It seems that sublist feature got implemented: https://github.com/gorhill/uBlock/commit/84fad1deebe5b453de6526cab7e93d06bc8baa84. Great!

However, I feel that the button to toggle sublists open is somewhat hard to notice at it's current location. It's easy to miss, unless you happen to know beforehand that the sublist feature exists.

Currently, it looks like this:

![kuva](https://user-images.githubusercontent.com/17256841/236699011-fb055e24-b8e3-439c-b179-71946b1b8073.png)

How about moving the sublist toggles to the left side of the checkbox? Then it would be easier to notice that a list has subsections:

![kuva](https://user-images.githubusercontent.com/17256841/236699100-637abbed-55de-4449-8890-3aba7c149749.png)

**Reply by gorhill** [reply]
I tried this originally, decided against it in the end.

**Reply by peace2000** [reply]
Or this way?

![kuva](https://user-images.githubusercontent.com/17256841/236829147-b68d6acc-a920-41eb-a896-3e89bd52e5ab.png)

Or do you feel that it's better not to have toggles near by the checkbox?

**Reply by aranapatona** [reply]
I also think Sublists arrow should be to the left, in consistency with the list main button, having them at the end of lines just makes them get lost.

**Comment by Yuki2718** [comment]
I found only new FA and not ELC or FS is enabled if old FA was enabled before the change. I guess all the three should be checked? I also wonder why Resource abuse shows 0 use...

![lists](https://user-images.githubusercontent.com/58900598/236770054-6017e458-3348-4d69-9b60-196d36b0a12f.png)

**Reply by MasterKia** [reply]
> why Resource abuse shows 0 use

Maybe because it was merged with Privacy?

**Comment by partingscientist** [comment]
I recently noticed that there's a sublist feature being implemented, which I'm able to use when installing the dev build in a new profile. However, the `Filter Lists` page on my dirty profile seems to be the same (i.e. no sublists) even though I'm also using the dev build. Did I miss some kind of migration step(s)?

I'm using uBlock Origin 1.49.3b9 on Firefox 113.0.1 on both profiles.

**Reply by ghost** [reply]
Will update under two weeks. Shift + click on "Purge all caches", then update, to force it.

**Comment by krystian3w** [comment]
Maybe search bar should be in same line like counter for PC?

![obraz](https://github.com/uBlockOrigin/uBlock-issues/assets/35370833/0a31a758-33d9-4676-becc-9fca08b6e832) |
--- |
![obraz](https://github.com/uBlockOrigin/uBlock-issues/assets/35370833/b3a39ca8-c5f8-437d-b173-48a07f7a404b) |

**Reply by krystian3w** [reply]
I can share `userContent.css` with many solutions to disable move to new line:

<ul> <li> <p>one - inline-block:</p> 

```css
#lists .rootstats { display: inline-block !important; }
```

[chrome.zip](https://github.com/uBlockOrigin/uBlock-issues/files/11470039/chrome.zip)

</ul>

**Comment by peace2000** [comment]
ryanbr has added a new newsletter subcategory:

https://github.com/easylist/easylist/tree/master/fanboy-addon

fanboy_newsletter_shopping_allowlist.txt
fanboy_newsletter_shopping_allowlist_general_hide.txt
fanboy_newsletter_shopping_specific_ABP.txt
fanboy_newsletter_shopping_specific_block.txt
fanboy_newsletter_shopping_specific_hide.txt	
fanboy_newsletter_shopping_specific_uBO.txt

These (not ABP) should be added to the Easylist Newsletter section as source files. Now they are not included.

fyi @gorhill 

Edit, or maybe I can add them by myself also...

**Reply by peace2000** [reply]
Added: https://github.com/uBlockOrigin/uAssets/commit/ce1fb870ea101df369343b1dc3e19313c8d71f2f

**Reply by gorhill** [reply]
~~It's all in the dev build, I will copy over `assets.dev.json` to `assets.json` when I publish stable release.~~ Oh ok, never mind.
